cluster/network.go

```diff
@@ -0,0 +1,59 @@
+package cluster
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/Sirupsen/logrus"
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/api/types/container"
+	"github.com/rancher/rke/docker"
+	"github.com/rancher/rke/k8s"
+	"github.com/rancher/rke/pki"
+)
+
+func (c *Cluster) DeployNetworkPlugin() error {
+	deployerHost := c.ControlPlaneHosts[0]
+	logrus.Debugf("[Network] Using host [%s] for deployment", deployerHost.Hostname)
+	logrus.Debugf("[Network] Pulling network-deployer image..")
+	if err := docker.PullImage(deployerHost.DClient, deployerHost.Hostname, k8s.KubectlImage); err != nil {
+		return err
+	}
+	logrus.Debugf("[Network] Setting network plugin: %s", c.NetworkPlugin)
+	env := []string{
+		fmt.Sprintf("%s=%s", "KUBECFG_NET_DEPLOY", c.Certificates[pki.KubeAdminCommonName].Config),
```

Comment:
can you separate the kubectl run in a separate function like (RunKubectl) in k8s package that can receive the cmd string as in `/tmp/entrypoint.sh apply -f /network/ ....` which should be more generic and just call this function from the deploy network plugin